assistbehaviour.lua

```diff
@@ -56,15 +56,15 @@ function AssistBehaviour:Update()
 		local uname = self.name
 		if self.isCommander then
 			-- turn commander into build assister if you control more than half the mexes or if it's damaged
-			if ai.nonAssistant[self.id] then
-				if (IsSiegeEquipmentNeeded() or unit:GetHealth() < unit:GetMaxHealth() * 0.9) and ai.factories ~= 0 and ai.conCount > 2 then
-					ai.nonAssistant[self.id] = nil
+			if self.ai.nonAssistant[self.id] then
+				if (self.ai.haveAdvFactory or IsSiegeEquipmentNeeded() or unit:GetHealth() < unit:GetMaxHealth() * 0.9) and self.ai.factories ~= 0 and self.ai.conCount > 2 then
```

Comment:
i would prefer not using functions from taskqueues in a module. i'm even sure how taskqueues is available here